packages/headless/src/controllers/query-summary/headless-query-summary.ts

```diff
@@ -32,6 +34,7 @@ export const buildQuerySummary = (
     get state() {
       const state = engine.state;
       return {
+        ...searchStatus.state,
```

Comment:
Could we just include the `hasResults` property instead of destructuring?

Adding everything from `searchStatus.state` will cause changes in any of the properties to invoke `subscribe` callbacks to rerender. The more state properties we expose on a single controller, the less effective JSON.stringify becomes.

Reply:
isLoading and isFirstSearchExecuted are both useful here, I just removed duplication of the hasResults